### PR TITLE
frontend: Adds a dropdown for disabling the creation of private r53 z…

### PR DESCRIPTION
**Pull request by kans**
…one when you select a public one.

### CODE REVIEW

**Comment by kans** [comment]
screen shot 2017-05-19 at 3 46 02 pm

**Comment by kans** [comment]
screen shot 2017-05-19 at 3 46 32 pm

File: installer/frontend/components/aws-vpc.jsx [hunk 1]
```diff
@@ -168,6 +172,8 @@ const stateToProps = ({aws, clusterConfig}) => {
     podCIDR: clusterConfig[POD_CIDR],
     serviceCIDR: clusterConfig[SERVICE_CIDR],
     advanced: clusterConfig[AWS_ADVANCED_NETWORKING],
+    privateZone: _.get(clusterConfig, ['extra', AWS_HOSTED_ZONE_ID, 'privateZones', clusterConfig[AWS_HOSTED_ZONE_ID]]),
+    privateZone1: ['extra', AWS_HOSTED_ZONE_ID, 'privateZones', clusterConfig[AWS_HOSTED_ZONE_ID]],
```
Comment (squat) [inline installer/frontend/components/aws-vpc.jsx]: is this used?

**Comment by meghanschofield** [comment]
hellooo! So, we had different wording for this - see below:

Default choice
screen shot 2017-05-22 at 10 57 54 am

Secondary choice:
screen shot 2017-05-22 at 10 57 57 am

The choices were written to match other similar options that live in and around the installer...consistency is helpful! Is there a reason these need to be edited?

File: installer/frontend/cluster-config.js [hunk 2]
```diff
@@ -277,12 +287,19 @@ export const toAWS_TF = (cc, FORMS, opts={}) => {
     ret.variables.tectonic_aws_external_worker_subnet_ids = workerSubnets;
     ret.variables.tectonic_aws_external_vpc_public = cc[AWS_CREATE_VPC] !== 'VPC_PRIVATE';
   }
+
+  const privateZone = _.get(cc, toExtraData(AWS_HOSTED_ZONE_ID) + '.privateZones.' + cc[AWS_HOSTED_ZONE_ID]);
+  if (!privateZone && cc[AWS_SPLIT_DNS] === splitDNS_off) {
+    ret.variables.tectonic_aws_external_private_zone = cc[AWS_HOSTED_ZONE_ID];
+  } else {
+    ret.variables.tectonic_aws_external_private_zone = '';
```
Comment (ggreer) [inline installer/frontend/cluster-config.js]: You don't need this `else`.

File: installer/frontend/__tests__/examples/aws-vpc.json [hunk 3]
```diff
@@ -48,6 +48,7 @@
     "tectonic_experimental": false,
     "tectonic_master_count": 1,
     "tectonic_service_cidr": "10.3.0.0/16",
-    "tectonic_worker_count": 3
+    "tectonic_worker_count": 3,
+    "tectonic_aws_external_private_zone": ""
```
Comment (ggreer) [inline installer/frontend/__tests__/examples/aws-vpc.json]: This goes away if you remove the `else` in clusterconfig.

**Comment by kans** [comment]
@meghanschofield 
Apologies, my understanding was that the text was placeholder.  I actually wanted to talk to you about it on Friday. I'd argue the text needs to be edited because "optimize DNS" has no technical meaning. You can't guess what the dropdown does without reading the small print.

**Comment by meghanschofield** [comment]
I worked pretty hard to get those words worked out. We need folks to know we are doing it on their behalf (hence the words, do this for me) or not. That is also similar to language used in the installer for similar things. 

Everyone I talked (like 5 folks) to didn't know what "split-view" was, or what we meant by "use public and private zones" because you make that choice with the radio buttons - so we went with Optimize DNS. 

Should we talk IRL!? I know we're moving fast, and this is a new thing. Maybe I still don't understand all the points here. :} We can rework the words together, live, if that helps @kans !

**Comment by kans** [comment]
@meghanschofield new PR is up at https://github.com/coreos/tectonic-installer/pull/836 .  Lets talk over there.

**Pull request merged**